packages/devtools_app/lib/src/timeline/timeline_flame_chart.dart

```diff
@@ -834,15 +842,15 @@ class AsyncGuidelinePainter extends FlameChartPainter {
   // TODO(kenz): does this have to return true all the time? Is it cheaper to
   // compare delegates or to just paint?
   @override
-  bool shouldRepaint(AsyncGuidelinePainter oldDelegate) => true;
+  bool shouldRepaint(CustomPainter oldDelegate) => true;
```

Comment:
why did this type change?

Reply:
You should avoid overriding a dart method with an implementation with a more specific type. Not sure why we don't have a lint to prevent it. This would throw an exception in a debug build if shouldRepaint was called with a CustomPainter of a different type.